### PR TITLE
Rework inter-module dependencies

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by dbmeneses** [comment]
LGTM, nice change

File: server/sonar-server/src/main/java/org/sonar/server/startup/JdbcDriverDeployer.java [hunk 1]
```diff
@@ -66,8 +67,12 @@ public void start() {
 
   private static String driverIndexContent(@Nullable File deployedDriver) {
     if (deployedDriver != null) {
-      String hash = new FileHashes().of(deployedDriver);
-      return deployedDriver.getName() + "|" + hash;
+      try (FileInputStream fis = new FileInputStream(deployedDriver)) {
+        String hash = DigestUtils.md5Hex(fis);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/startup/JdbcDriverDeployer.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2070)

File: sonar-core/src/main/java/org/sonar/core/platform/RemotePlugin.java [hunk 2]
```diff
@@ -64,7 +66,11 @@ public RemotePlugin setFile(String filename, String hash) {
   }
 
   public RemotePlugin setFile(File f) {
-    return this.setFile(f.getName(), new FileHashes().of(f));
+    try (FileInputStream fis = new FileInputStream(f)) {
+      return this.setFile(f.getName(), DigestUtils.md5Hex(fis));
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/platform/RemotePlugin.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2070)

File: server/sonar-server/src/main/java/org/sonar/server/batch/BatchIndex.java [hunk 3]
```diff
@@ -56,7 +56,11 @@ public void start() {
       for (File file : files) {
         String filename = file.getName();
         if (StringUtils.endsWith(filename, ".jar")) {
-          sb.append(filename).append('|').append(new FileHashes().of(file)).append(CharUtils.LF);
+          try (FileInputStream fis = new FileInputStream(file)) {
+            sb.append(filename).append('|').append(DigestUtils.md5Hex(fis)).append(CharUtils.LF);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/batch/BatchIndex.java]: CRITICAL Use a stronger hashing algorithm than MD5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2070)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* CRITICAL 3 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**